### ISSUE TITLE
segfault when HTTP request returned status code 400 during sync / resync process

### ISSUE THREAD

**Issue by abraunegg** [skilion/onedrive#75]
If manually removing the App privilege from account.live.com, the onedrive client fails to handle that it is no longer authorised to sync - leading to improper handling of receiving a http 400 error code:

```
Deleting the saved status ...
Loading config ...
Initializing the OneDrive API ...
Opening the item database ...
All operations will be performed in: /root/OneDrive
Initializing the Synchronization Engine ...
Applying differences ...
HTTP request returned status code 400 (Bad Request)
Applying differences ...
HTTP request returned status code 400 (Bad Request)
Applying differences ...
sync.SyncException@src/sync.d(106): HTTP request returned status code 400 (Bad Request)
----------------
??:? void sync.SyncEngine.applyDifferences() [0x599b99]
??:? void main.performSync(sync.SyncEngine) [0x58b487]
??:? _Dmain [0x58abf6]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFZv [0x5a58ee]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5a5838]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x5a58aa]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5a5838]
??:? _d_run_main [0x5a57a9]
??:? main [0x58d511]
??:? __libc_start_main [0xfae20d1c]
onedrive.OneDriveException@src/onedrive.d(342): HTTP request returned status code 400 (Bad Request)
----------------
??:? std.json.JSONValue onedrive.OneDriveApi.post!(char).post(const(char)[], const(char)[]) [0x5953b1]
??:? void onedrive.OneDriveApi.checkAccessTokenExpired() [0x5910da]
??:? void sync.SyncEngine.applyDifferences() [0x599840]
??:? void main.performSync(sync.SyncEngine) [0x58b487]
??:? _Dmain [0x58abf6]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFZv [0x5a58ee]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5a5838]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x5a58aa]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5a5838]
??:? _d_run_main [0x5a57a9]
??:? main [0x58d511]
??:? __libc_start_main [0xfae20d1c]
Segmentation fault
```

Perhaps a check is needed to validate the saved credentials before proceeding with the sync / monitor processes. The only way to 're-auth' the client in this situation (manually removing authorization in the live account) is to remove all the  files under ~/.config/onedrive/

**Comment by Pogodaanton** [comment]
Thanks for telling a solution for that, I've changed my password and I couldn't figure out a solution to redo the authorization process. Maybe, if onedrive is uninstalled, it should also remove the local config files.

**Comment by Miboa** [comment]
What I did was delete "refresh_token" and "status_token" in Home/.config/onedrive  and then run onedrive again.

**Comment by skilion** [comment]
Fixed in ad4e910e5584f584e271e5ab6a52c9daacedb178
